components/Article/Page.js

```diff
@@ -471,50 +469,22 @@ class ArticlePage extends Component {
       showSeriesNav
     } = this.state
 
-    const hasPdf = meta && meta.template === 'article'
     const isEditorialNewsletter =
       meta && meta.template === 'editorialNewsletter'
-    const actionBar = meta && (
-      <ArticleActionBar
-        t={t}
-        url={meta.url}
-        title={meta.title}
-        animate={!podcast}
-        template={meta.template}
-        path={meta.path}
-        showShare={
-          !isEditorialNewsletter || (newsletterMeta && newsletterMeta.free)
-        }
-        linkedDiscussion={meta.linkedDiscussion}
-        ownDiscussion={meta.ownDiscussion}
-        dossierUrl={meta.dossier && meta.dossier.meta.path}
-        onAudioClick={meta.audioSource && this.toggleAudio}
-        onGalleryClick={meta.indicateGallery ? this.showGallery : undefined}
-        onPdfClick={
-          hasPdf && countImages(article.content) > 0
-            ? this.togglePdf
-            : undefined
-        }
-        pdfUrl={hasPdf ? getPdfUrl(meta) : undefined}
-        documentId={article.id}
-        repoId={article.repoId}
-        isEditor={isEditor}
-        userBookmark={article.userBookmark}
-        userProgress={article.userProgress}
-        showBookmark={isMember}
-        estimatedReadingMinutes={meta.estimatedReadingMinutes}
-        estimatedConsumptionMinutes={meta.estimatedConsumptionMinutes}
-        subscriptions={article.subscribedBy.nodes}
-        showSubscribe
-        isDiscussion={meta && meta.template === 'discussion'}
-      />
+
+    const actionBar = article && (
+      <ActionBar mode='article-top' document={article} />
     )
     const actionBarEnd = actionBar
       ? React.cloneElement(actionBar, {
-          animate: false,
-          estimatedReadingMinutes: undefined,
-          estimatedConsumptionMinutes: undefined,
-          grandSharing: !inNativeApp
```

Comment:
Would it be hard to bring back `grandSharing` at the end of an article? This currently causes the following share buttons at the end:
<img width="716" alt="Screenshot 2020-08-27 at 16 28 35" src="https://user-images.githubusercontent.com/410211/91455544-7dab6b00-e882-11ea-8090-5b324a625f3e.png">
 
(not shown in the app for consistency because share triggers the native share dialog there)

We seem to already have a similar display with the new `IconButton` on the marketing page:
<img width="706" alt="Screenshot 2020-08-27 at 16 27 59" src="https://user-images.githubusercontent.com/410211/91455666-a895bf00-e882-11ea-8ae5-b0d86952e7d0.png">

Could we use that underneath the article too?

Reply:
jup, shouldn't be a big change.